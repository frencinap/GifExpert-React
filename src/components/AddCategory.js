import React, { useState } from "react";
import PropTypes from 'prop-types'

//recibimos setCategories
export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState( '' )

    const handleInputValue = (e) => {
        setInputValue( e.target.value )
    }

    const handleSubmit = (e) => {
        //prevenir el comportamiento por defecto del submit
        e.preventDefault()
        //console.log('Submit Hecho');
        //validación submit
        if( inputValue.trim().length > 2) {
            setCategories( cats => [...cats, inputValue] )
            setInputValue('')
        }     
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputValue }
                placeholder="Agrega un héroe"
            />
        </ form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}