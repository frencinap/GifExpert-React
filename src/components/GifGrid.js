import React from 'react'

export const GifGrid = ({ category }) => {

    const getGif = async() =>{
        
        const url = 'https://api.giphy.com/v1/gifs/search?q=Avengers&limit=10&api_key=kQ50E0uTSpq7EGd7Nau8q6hA3EgxiN1P'
        const resp = await fetch( url )
        const { data } = await resp.json()

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGif()
    return (
        <div>
            <h3> { category } </h3>
        </div>
    )
}
