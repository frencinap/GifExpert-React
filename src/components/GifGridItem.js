import React from 'react'
import PropTypes from 'prop-types';

//desestructuradas gracias al spreadperator de GifGrid
export const GifGridItem = ( {id, title, url} ) => {

    console.log( id, title, url );
    return (
        <div className='card animate__animated animate__backInUp'>
            <img src={ url } alt={title} />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired

}