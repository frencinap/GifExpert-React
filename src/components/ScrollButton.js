import React, { useState } from "react";
import { FiArrowUpCircle } from "react-icons/fi";
import { IconContext } from "react-icons";

export const ScrollButton = () => {

    const [visible, setVisible] = useState(false)

    const toggleVisible = () => {
        const scrolled = document.documentElement.scrollTop
        if( scrolled > 150 ){
            setVisible(true)
        }
        else if ( scrolled <= 150 ){
            setVisible(false)
        }

    }

    const scrollToTop = () => {
        window.scrollTo({
            top: 0,
            behavior: 'smooth',
        })
    }

    window.addEventListener('scroll', toggleVisible)

    return (
        <>
            <IconContext.Provider value={{ className: "arrow"}}>
                <FiArrowUpCircle
                style={{display: visible ? 'inline' : 'none'}}
                onClick={ scrollToTop } />
            </IconContext.Provider>
        </>    
    )
};
