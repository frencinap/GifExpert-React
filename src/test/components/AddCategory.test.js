import React from "react";
import { shallow } from "enzyme"
import { AddCategory } from '../../components/AddCategory'
import '@testing-library/jest-dom'


describe('pruebas en el componente AddCategory', () => {

    //funcion vacia para simular setCategories
    const setCategories = jest.fn()
    let wrapper = shallow( < AddCategory setCategories={ setCategories } />)

    //limpiamos los campos antes de cada prueba
    beforeEach(()=>{
        jest.clearAllMocks()
        wrapper = shallow( < AddCategory setCategories={ setCategories } />)
    })

    test('debe hacer el match con el snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot()
        
    });

    test('debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input')
        const value = 'Hola mundo'
        //los eventos onAlgo se simulan si el on( en este caso onChange)
        input.simulate('change', { target: { value } } )
        
    });

    test('No debe de postear la información con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} })
        //como el submit está vacio setCategories no debe ser llamado
        expect( setCategories ).not.toHaveBeenCalled()

    });

    test('Se debe llamar el setCategories y limpiar la caja de texto', () => {

        const value = 'Hola tarola'
        //simular inputChange
        wrapper.find('input').simulate('change', { target: { value }})
        //simluacion submit
        wrapper.find('form').simulate('submit', { preventDefault(){} } )
        //setCategories fue llamado
        expect( setCategories ).toHaveBeenCalled()
        //fue llamado con una función
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) )
        //el valor del input debe quedar vacío
        expect( wrapper.find('input').prop('value')).toBe('')
  
    });
    
});