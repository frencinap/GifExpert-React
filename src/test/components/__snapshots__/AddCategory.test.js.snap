// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente AddCategory debe hacer el match con el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Busca un hÃ©roe"
    type="text"
    value=""
  />
</form>
`;
